### ISSUE TITLE
The CA1062 is shown for extension methods even with .editorconfig file with exclude_extension_method_this_parameter = true

### ISSUE THREAD

**Issue by OldrichDlouhy** [dotnet/roslyn-analyzers#4046]
The CA1062 is shown for extension methods even with `.editorconfig` file with `dotnet_code_quality.CA1062.exclude_extension_method_this_parameter = true`

### Steps To Reproduce

Have .NET Core 3.0.1 class library project with Microsoft.CodeAnalysis.FxCopAnalyzers 3.3.0 with following files:

`.editorconfig`
```
###############################
# Code Analysis settings      #
###############################
dotnet_code_quality.CA1062.exclude_extension_method_this_parameter = true
```

`DictionaryExtensions.cs`
``` CSharp
using System.Collections.Generic;

namespace Tools
{
    public static class DictionaryExtensions
    {
        public static TValue TryGet<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key)
        {
            dictionary.TryGetValue(key, out TValue value);
            return value;
        }
    }
}
```

### Expected behavior

No CA1062 warning is shown on compilation/code analysis

### Actual behavior

CA 1062 warning is shown:
```
warning CA1062: In externally visible method 'TValue DictionaryExtensions.TryGet<TKey, TValue>(IDictionary<TKey, TValue> dictionary, TKey key)', validate parameter 'dictionary' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.
```

### Analyzer

**Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.0 (Latest)

**Diagnostic ID**: [CA1062](https://docs.microsoft.com/visualstudio/code-quality/ca1062)

## Additional context

In previous version of Microsoft.CodeAnalysis.FxCopAnalyzers (3.0.0) the CA1062 was correctly suppressed for the extension methods.

**Comment by OldrichDlouhy** [comment]
Thanks for the response.

Adding the `[*.cs]` header to the `.editorconfig` file fixed the issue and the CA1062 warning is not shown for extension method.

**Comment by Evangelink** [comment]
@mavasani As far as I can see there is nothing to do for this ticket, except if you do want to handle entries without the language specified?

**Comment by mavasani** [comment]
Yes, thanks. Closing out.

**Comment by tlunsfordCXP** [comment]
ICIH, I found that vscode required a window reload in order to pick up the `exclude_extension_method_this_parameter` editorconfig setting.
